packages/charts/src/components/BarChart/BarRechart.tsx

```diff
@@ -0,0 +1,140 @@
+      secondYAxis: {
+        dataKey: '',
+        name: '',
+        color: 'black'
+      }
+    }
+  } = props;
+
+  useInitialize();
+
+  const { parameters }: any = useTheme();
+  const chartRef = useConsolidatedRef<any>(ref);
+
+  const dataKeys = useMemo(() => (dataset ? Object.keys(dataset[0]).filter((key) => key !== labelKey) : []), [dataset]);
+  const colorSecondY = useMemo(
+    () => (chartConfig.secondYAxis ? dataKeys.findIndex((key) => key === chartConfig.secondYAxis.dataKey) : 0),
+    [dataset]
+  );
+
+  const onItemLegendClick = useCallback(
+    (e) => {
+      if (legendClickHandler) {
+        legendClickHandler({
+          dataKey: e.dataKey,
+          value: e.value,
+          chartType: e.type,
+          color: e.color,
+          payload: e.payload
+        });
+      }
+    },
+    [dataset]
+  );
+
+  const onDataPointClick = useCallback(
+    (e) => {
+      if (e && dataPointClickHandler) {
+        dataPointClickHandler({
+          dataKey: Object.keys(e).filter((key) => e[key] === e.value && key !== 'value')[0],
+          value: e.value,
+          payload: e.payload
+        });
+      }
+    },
+    [dataset]
+  );
+
+  return (
+    <ChartContainer dataset={dataset} loading={loading} placeholder={BarChartPlaceholder} width={width} height={height}>
+      <BarChartLib
+        ref={chartRef}
+        data={dataset}
+        style={{ fontSize: parameters.sapUiFontSmallSize }}
+        barGap={chartConfig.barGap}
+      >
+        <CartesianGrid
+          vertical={chartConfig.gridVertical}
+          horizontal={chartConfig.gridHorizontal}
+          stroke={chartConfig.gridStroke}
+        />
+        {(chartConfig.xAxisVisible === true || chartConfig.xAxisVisible === undefined) && <XAxis dataKey={labelKey} />}
+        {(chartConfig.yAxisVisible === true || chartConfig.yAxisVisible === undefined) && <YAxis yAxisId="left" />}
+        {chartConfig.secondYAxis && (
+          <YAxis
+            dataKey={chartConfig.secondYAxis.dataKey}
+            stroke={
+              chartConfig.secondYAxis.color
+                ? chartConfig.secondYAxis.color
+                : `var(--sapUiChartAccent${(colorSecondY % 12) + 1})`
+            }
+            label={{ value: chartConfig.secondYAxis.name, angle: +90, position: 'center' }}
+            orientation="right"
+            yAxisId="right"
+          />
+        )}
+        {dataKeys.map((key, index) => (
+          <Bar
+            yAxisId={chartConfig.secondYAxis && chartConfig.secondYAxis.dataKey === key ? 'right' : 'left'}
+            stackId={chartConfig.stacked ? 'A' : undefined}
+            strokeOpacity={chartConfig.strokeOpacity}
+            fillOpacity={chartConfig.fillOpacity}
+            label={chartConfig.dataLabel && { position: 'top', fontFamily: parameters.sapUiFontFamily }}
+            key={key}
+            name={key}
+            dataKey={key}
+            fill={color ? color : `var(--sapUiChartAccent${(index % 12) + 1})`}
```

Comment:
optional chaining might fit here as well
```suggestion
            fill={color ?? `var(--sapUiChartAccent${(index % 12) + 1})`}
```